src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.html

```diff
@@ -162,15 +168,78 @@ <h5><span jhiTranslate="artemisApp.studentExamDetail.exercises">Exercises</span>
                                 *ngIf="
                                     exercise?.studentParticipations &&
                                         exercise?.studentParticipations.length > 0 &&
-                                        (exercise?.studentParticipations)![0].results &&
+                                        (exercise?.studentParticipations)![0].results?.length > 0 &&
                                         (exercise?.studentParticipations)![0].results[0].assessor;
                                     else noAssessor
                                 "
                             >
-                                {{ (exercise?.studentParticipations)![0].results[0].assessor.name }}
+                                {{ (exercise?.studentParticipations)![0].results[0].assessor?.name }}
                             </ng-container>
                             <ng-template #noAssessor></ng-template>
                         </td>
+                        <td class="align-middle">
+                            <span
+                                *ngIf="
+                                    exercise?.studentParticipations?.length > 0 &&
+                                    (exercise?.studentParticipations)![0].results?.length > 0 &&
```

Comment:
You do this check a lot so it might be better to move this to a separate component that definitely has and exercise and results etc. It will cause problems with strict templates as well.

Reply:
Great catch, done!